### ISSUE TITLE
Custom `g:targets_aiAI` breaks `cina` when next argument is empty

### ISSUE THREAD

**Issue by balta2ar** [wellle/targets.vim#150]
I wanted to swap the effects of `i` and `I` thus I added this:
``` vim
let g:targets_aiAI = 'aIAi'
```
Unfortunately, it breaks `cina` when next argument is empty. For example I have this line:
```
function(fir|st, )
```
I press `cina` and the cursor stays on the same character and enters Insert mode. Without `targets_aiAI` it's working as expected: cursor moves after the comma and enters Insert mode. Interestingly, `cIna` is still working with custom `targets_aiAI`. @wellle Am I missing something?

**Comment by wellle** [comment]
I don't think this is related to the custom setting. The same thing happens when you type `cIna` with default settings. The modifier `I` means remove all surrounding whitespace. But it fails if there is only whitespace. That's because it's not clear what empty match should be selected.

In your example the next argument is slightly malformed because, strictly speaking, there is no real argument to operate on.

For this specific case we could consider removing all whitespace to operate on the empty match after the space. But I'm not sure that would lead to expected behavior in other cases.

**Comment by balta2ar** [comment]
Thank you for the prompt reply! And yes, you're right... `I` fails without custom setting... I wish I could have both of two worlds, because now I have to do `cina` with default settings and add space after a comma which is super annoying.

**Comment by wellle** [comment]
Yeah I understand your problem. I opened a small WIP branch  [150-wip](https://github.com/wellle/targets.vim/compare/150-wip) to address this very specific use case you presented. But it still fails if there are no spaces at all like `f(,x)` and `g(x,)`. In order to fix those as well we would need make `I` work with targets that don't contain any spaces at all.

**Comment by nkgm** [comment]
Just like @balta2ar, it makes more sense for me to use:
```VimL
let g:targets_aiAI = 'aIAi'
```
The problem then is that built-in vim text-objects are also affected. For example `viw` now doesn't work - instead I have to use `vIw`.
